static/js/redux/ui/search_bar.jsx

```diff
@@ -13,185 +13,204 @@ GNU General Public License for more details.
 */
 
 import PropTypes from 'prop-types';
-import React from 'react';
+import React, { useState, useEffect, useRef, useCallback } from 'react';
 import ClickOutHandler from 'react-onclickout';
 import classNames from 'classnames';
 import SearchSideBarContainer from './containers/search_side_bar_container';
 import * as SemesterlyPropTypes from '../constants/semesterlyPropTypes';
 import SearchResult from './search_result';
 
-class SearchBar extends React.Component {
+/* eslint-disable react/no-unused-prop-types */
```

Comment:
Why do you disable the eslint/proptypes rule? I never used this myself, so I thought I'd ask 👍.

Reply:
There is a false positive issue with functional components. Props used inside a function that resides within the component will be marked as unused by ESlint. 
On the [documentation](https://github.com/yannickcr/eslint-plugin-react/blob/master/docs/rules/no-unused-prop-types.md#false-positives-sfc) they said:
`Stateless Function Components (SFCs) accept props as an argument and return a JSX expression. Even if the function gets called from a component, the props that are only used inside the component are not be considered used by a component.
`
They did offer a suggestion which is to assign another variable that equals props and pass that into the inner function as a parameter. This would work too but I thought It might add extra complexity, so I disabled the rule. Please let me know what do you think?